community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NumberArrayFactory.java

```diff
@@ -212,60 +213,60 @@ public Auto( NumberArrayFactory... candidates )
         @Override
         public LongArray newLongArray( long length, long defaultValue, long base )
         {
-            OutOfMemoryError error = null;
+            Throwable error = null;
             for ( NumberArrayFactory candidate : candidates )
             {
                 try
                 {
                     return candidate.newLongArray( length, defaultValue, base );
                 }
-                catch ( OutOfMemoryError e )
+                catch ( Throwable e )
                 {   // Allright let's try the next one
                     error = e;
                 }
             }
-            throw error( length, 8, error );
+            throw launderedException( error( length, 8, error ) );
```

Comment:
The error message produced by `error` seem to be a bit off with the new behaviour. It claims "Not enough memory available for allocating..." which is a bit misleading in the case of an `ArithmeticException`.
I would also like us to have a more generic message in the event of some other unexpected exception.